Source/WebGPU/WebGPU/Device.mm

```diff
@@ -582,15 +582,17 @@ static void setOwnerWithIdentity(id<MTLResourceSPI> resource, auto webProcessID)
         options.fastMathEnabled = YES;
         ALLOW_DEPRECATED_DECLARATIONS_END
         /* NOLINT */ id<MTLLibrary> library = [m_device newLibraryWithSource:@R"(
+#define vertexCount 0
+#define primitiveRestart 1
     using namespace metal;
-    [[vertex]] void vsUshort(device const ushort* indexBuffer [[buffer(0)]], device MTLDrawIndexedPrimitivesIndirectArguments& indexedOutput [[buffer(1)]], const constant uint* vertexCount [[buffer(2)]], uint indexId [[vertex_id]]) {
-        ushort plusOneValue = 1 + indexBuffer[indexId]; // plusOne to handle primitive restarts
-        if (indexedOutput.baseVertex >= vertexCount[0] || plusOneValue + indexedOutput.baseVertex >= vertexCount[0] + 1)
+    [[vertex]] void vsUshort(device const ushort* indexBuffer [[buffer(0)]], device MTLDrawIndexedPrimitivesIndirectArguments& indexedOutput [[buffer(1)]], const constant uint* data [[buffer(2)]], uint indexId [[vertex_id]]) {
+        ushort plusOneValue = data[primitiveRestart] + indexBuffer[indexId]; // plusOne to handle primitive restarts
+        if (indexedOutput.baseVertex >= data[vertexCount] || plusOneValue + indexedOutput.baseVertex >= data[vertexCount] + data[primitiveRestart])
             indexedOutput.indexCount = 0u;
     }
-    [[vertex]] void vsUint(device const uint* indexBuffer [[buffer(0)]], device MTLDrawIndexedPrimitivesIndirectArguments& indexedOutput [[buffer(1)]], const constant uint* vertexCount [[buffer(2)]], uint indexId [[vertex_id]]) {
-        uint plusOneValue = 1 + indexBuffer[indexId]; // plusOne to handle primitive restarts
-        if (indexedOutput.baseVertex >= vertexCount[0] || plusOneValue + indexedOutput.baseVertex >= vertexCount[0] + 1)
+    [[vertex]] void vsUint(device const uint* indexBuffer [[buffer(0)]], device MTLDrawIndexedPrimitivesIndirectArguments& indexedOutput [[buffer(1)]], const constant uint* data [[buffer(2)]], uint indexId [[vertex_id]]) {
+        uint plusOneValue = data[primitiveRestart] + indexBuffer[indexId]; // plusOne to handle primitive restarts
```

Comment:
ditto

Reply:
oh indeed, I will correct this, thanks for catching